File: kata/almost_binary.java

import java.lang.Math;
import java.util.Scanner;
class Solution {
  public static void main(String args[]) {
    Scanner sc = new Scanner(System.in);
    String in;
     do {
        in = sc.nextLine();
        int sum = 0;
        for(int i = 0; i< in.length(); i++) {
            if(in.charAt(i) == '#') {
                sum += (int) Math.pow(2, (in.length() - i - 1));
            }
        }
        System.out.println(sum);
    } while(sc.hasNextLine());
    
  }
};

File: kata/almost_binary.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		data := strings.Split(scanner.Text(), "")
		dlen := len(data)
		sum := 0

		for i, v := range data {
			if v == "#" {
				sum += int(math.Pow(float64(2), float64(dlen-i-1)))
			}
		}
		fmt.Println(sum)

	}
}
